Apps/Database/src/Delegates/DbProfileDelegate.cs

```diff
@@ -349,6 +349,14 @@ public DbResult<IEnumerable<UserProfileHistory>> GetUserProfileHistories(string
             return result;
         }
 
+        /// <inheritdoc/>
+        public async Task<int> GetClosedUserProfileCount(CancellationToken ct)
+        {
+            return await this.dbContext.UserProfileHistory
+                .Where(h => h.Operation == "DELETE" && !this.dbContext.UserProfile.Any(p => p.HdId == h.HdId))
```

Comment:
In a raw SQL method I would perform a outer join on UserProfile table and take all records where `UserProfile.UserProfileId` values are null?

would a where clause like this translate into something similar or would this become a subquery run per userprofile entry?

Reply:
The current query should generate SQL like this:

```
SELECT count(*)::int
FROM gateway."UserProfileHistory" AS u
WHERE u."Operation" = 'DELETE' AND NOT EXISTS (
    SELECT 1
    FROM gateway."UserProfile" AS u0
    WHERE u0."UserProfileId" = u."UserProfileId")
```

It is possible to generate SQL like this:

```
SELECT count(*)::int
FROM gateway."UserProfileHistory" AS u
LEFT JOIN gateway."UserProfile" AS u0 ON u."UserProfileId" = u0."UserProfileId"
WHERE u."Operation" = 'DELETE' AND u0."UserProfileId" IS NULL
```

But it requires much messier LINQ:

```
return await
    this.dbContext.UserProfileHistory.Where(h => h.Operation == "DELETE")
        .GroupJoin(
            this.dbContext.UserProfile,
            history => history.HdId,
            profile => profile.HdId,
            (history, profiles) => new { history, profiles })
        .SelectMany(
            t => t.profiles.DefaultIfEmpty(),
            (t, existingProfile) => new { t, existingProfile })
        .Where(t => t.existingProfile == null)
        .Select(t => t.t.history)
        .CountAsync(ct);
```

And (as best I can tell), the execution plans are the same.